svir/dialogs/viewer_dock.py

```diff
@@ -284,6 +293,80 @@ def set_output_type_and_its_gui(self, new_output_type):
         # else.
         self.output_type = new_output_type
 
+    def load_agg_curves(self, calc_id, session, hostname, output_type):
+        self.change_output_type(output_type)
+        url = '%s/v1/calc/%s/extract/%s' % (hostname, calc_id, output_type)
+        resp_content = session.get(url).content
+        self.agg_curves = numpy.load(io.BytesIO(resp_content))
```

Comment:
As more outputs will become available as npz, you will probably want an utility function `extract_npz(session, hostname, calc_id, output_type)`.

Reply:
I agree. I will make some refactoring in order to load different kinds of outputs. I will also probably re-arrange the classification and management of different kinds of output types.

Reply:
I've done some refactoring to have extract_npz as a general method to be used by different loaders.